packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -139,39 +141,70 @@ export const runningSessionsStatus: JupyterFrontEndPlugin<void> = {
   description: 'Add the running sessions and terminals status bar item.',
   autoStart: true,
   requires: [IStatusBar, ITranslator],
+  optional: [ISettingRegistry],
   activate: (
     app: JupyterFrontEnd,
     statusBar: IStatusBar,
-    translator: ITranslator
+    translator: ITranslator,
+    settingRegistry: ISettingRegistry | null
   ) => {
-    const item = new RunningSessions({
-      onClick: () => app.shell.activateById('jp-running-sessions'),
-      onKeyDown: (event: KeyboardEvent<HTMLImageElement>) => {
-        if (
-          event.key === 'Enter' ||
-          event.key === 'Spacebar' ||
-          event.key === ' '
-        ) {
-          event.preventDefault();
-          event.stopPropagation();
-          app.shell.activateById('jp-running-sessions');
-        }
-      },
-      serviceManager: app.serviceManager,
-      translator
-    });
+    const createStatusItem = () => {
+      const item = new RunningSessions({
+        onClick: () => app.shell.activateById('jp-running-sessions'),
+        onKeyDown: (event: KeyboardEvent<HTMLImageElement>) => {
+          if (
+            event.key === 'Enter' ||
+            event.key === 'Spacebar' ||
+            event.key === ' '
+          ) {
+            event.preventDefault();
+            event.stopPropagation();
+            app.shell.activateById('jp-running-sessions');
+          }
+        },
+        serviceManager: app.serviceManager,
+        translator
+      });
+
+      item.model.sessions = Array.from(
+        app.serviceManager.sessions.running()
+      ).length;
+      item.model.terminals = Array.from(
+        app.serviceManager.terminals.running()
+      ).length;
+
+      return item;
+    };
 
-    item.model.sessions = Array.from(
-      app.serviceManager.sessions.running()
-    ).length;
-    item.model.terminals = Array.from(
-      app.serviceManager.terminals.running()
-    ).length;
+    const registerItem = () => {
+      const item = createStatusItem();
+      return statusBar.registerStatusItem(runningSessionsStatus.id, {
+        item,
+        align: 'left',
+        rank: 0
+      });
+    };
 
-    statusBar.registerStatusItem(runningSessionsStatus.id, {
-      item,
-      align: 'left',
-      rank: 0
-    });
+    if (settingRegistry) {
+      let disposable: IDisposable;
+      const onSettingsUpdated = (
+        settings: ISettingRegistry.ISettings
+      ): void => {
+        const showStatusBarItem = settings.get('showStatusBarItem')
+          .composite as boolean;
+        if (showStatusBarItem) {
+          disposable = registerItem();
```

Comment:
Wouldn't it register the item multiple times once we add more settings? I think it needs a guard to check if the disposable is defined and not disposed

Reply:
It looks like that would indeed be the case. Maybe we can add indeed add such guard to anticipate the case where new settings would be added 👍

Reply:
Or maybe we can just dispose the item, and recreate it if `showStatusBarItem` is `true`.